gooddata-java/src/main/java/com/gooddata/sdk/service/gdc/DataStoreService.java

```diff
@@ -95,7 +99,10 @@ public void upload(String path, InputStream stream) {
 
     private void upload(URI url, InputStream stream) {
         try {
-            sardine.put(url.toString(), stream);
+            // We need to use it this way, if we want to track request_id in the stacktrace.
+            InputStreamEntity entity = new InputStreamEntity(stream, -1);
+            List<Header> headers = Collections.singletonList(new BasicHeader(HTTP.EXPECT_DIRECTIVE, HTTP.EXPECT_CONTINUE));
+            ((SardineImpl) sardine).put(url.toString(), entity, headers, new ValidatingResponseHandler());
```

Comment:
what if it is not `SardineImpl`? Should it be handled somehow? I suppose ClassCastException is not something one would expect from calling the upload...

Reply:
I think this is not that case, where you should check for `ClassCastException`, because `Sardine` type has two inheritors where `SardineImpl` is one of them and we are instantiating their respective subclass types.

Reply:
so in the end it's always `GdcSardine` ... so why not to change the type of the field instaed of this casting?

Reply:
Changed to `GdcSardine`.